apps/wallet/src/ui/app/components/transactions-card/Transaction.tsx

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getExecutionStatusType,
+    getTransactionKindName,
+    getMoveCallTransaction,
+    getExecutionStatusError,
+    getTransferObjectTransaction,
+} from '@mysten/sui.js';
+import { useMemo } from 'react';
+import { Link } from 'react-router-dom';
+
+import { TxnTypeLabel } from './TxnActionLabel';
+import { TxnIcon } from './TxnIcon';
+import { TxnImage } from './TxnImage';
+import { CoinBalance } from '_app/shared/coin-balance';
+import { Text } from '_app/shared/text';
+import { getEventsSummary, getAmount, formatDate } from '_helpers';
+import { useMiddleEllipsis } from '_hooks';
+import { getTxnEffectsEventID } from '_redux/slices/txresults';
+
+import type { SuiTransactionResponse, SuiAddress } from '@mysten/sui.js';
+
+const TRUNCATE_MAX_LENGTH = 8;
+const TRUNCATE_PREFIX_LENGTH = 4;
+
+export function Transaction({
+    txn,
+    address,
+}: {
+    txn: SuiTransactionResponse;
+    address: SuiAddress;
+}) {
+    const { certificate } = txn;
+    const executionStatus = getExecutionStatusType(txn) as 'Success' | 'Failed';
```

Comment:
it looks like this returns `success` or `failure` not sure why we're casting this value

Reply:
you are right, removing